### PR TITLE
Tidy up transaction.js

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Tidy up transaction.js

## What does this PR address?
This PR cleans up transaction.js halving the lines of copy-pasted code and removing unused functions

## How does this benefit users?
The code is easier to audit and edit

### CODE REVIEW

**Review by panleone — DISMISSED** [review]
tACK 592e01724785cf4b0d98def5cbad358627e4f9a0:
normal tx delegation undelegation and MN creation working as expected

File: scripts/transactions.js [hunk 1]
```diff
@@ -306,34 +135,162 @@ export function delegateGUI() {
...
+            6000
+        );
+    }
+    // Verify the amount
+    const nAmount = Math.round(
+        Number(doms.domGuiUndelegateAmount.value.trim()) * COIN
+    );
+    if (!validateAmount(nAmount)) return;
+    const [address] = await getNewAddress();
+    //askForCSAddr(true);
```
Comment (JSKitty) [inline scripts/transactions.js:163]: This line can be nuked.
```suggestion
```

File: scripts/transactions.js [hunk 2]
```diff
@@ -306,34 +135,162 @@ export function delegateGUI() {
...
+    if (!result.ok && result.err === 'No change addr') {
+        askForCSAddr(true);
+        await undelegateGUI();
+    }
+}
+
+/**
+ * Creates and sends a transaction to the network.
+ * @param {Object} options
+ * @param {string} options.address - base38 encoded address to send funds to
```
Comment (JSKitty) [inline scripts/transactions.js:181]: base58
```suggestion
 * @param {string} options.address - base58 encoded address to send funds to
```

**Review by JSKitty — DISMISSED** [review]
**tACK**, tested with over 1000 UTXOs in many different transaction styles (Send - Delegate - Undelegate) :tada:

Extremely clean - great to finally have our last messy part of 'coin control' centralised in to a single transaction function finally, massive code reduction there alone.

Great job! :rocket: :fire:

**Review by JSKitty — APPROVED** [review]
**re-tACK** 8e6bfd5ff71957fb789b81cc357a6c0d773e22e5 - Perfect! :tada:

**Review by panleone — APPROVED** [review]
(re) tACK 8e6bfd5ff71957fb789b81cc357a6c0d773e22e5

**Pull request merged**